janusgraph-test/src/test/java/org/janusgraph/graphdb/query/QueryTest.java

```diff
@@ -158,6 +160,30 @@ public void testIdLookUpQuery() {
         assertTrue(tx.traversal().V().hasId(vId).has("pos", 0).hasNext());
         assertEquals(tx.traversal().V(vId).next(), tx.traversal().V().hasId(vId).next());
         assertEquals(tx.traversal().V(vId).next(), tx.traversal().V(vId).hasId(vId).next());
+
+        // vertices lookup followed by limit step
+        assertEquals(1, tx.traversal().V(vId, v2Id).limit(1).count().next());
+        assertEquals(1, tx.traversal().V(vId, v2Id).limit(1).toList().size());
+        assertEquals(1, tx.traversal().V().hasId(vId, v2Id).limit(1).count().next());
+        assertEquals(0, tx.traversal().V().hasId(vId, v2Id).limit(0).count().next());
+        assertEquals(1, tx.traversal().V().has(T.id, P.within(vId, v2Id)).limit(1).count().next());
+        assertEquals(0, tx.traversal().V().has(T.id, P.within(vId, v2Id)).limit(0).count().next());
+
+        // vertices lookup followed by order step
+        assertEquals(Arrays.asList(20, 30), tx.traversal().V().order().by("age").values("age").toList());
+        assertEquals(Arrays.asList(20, 30), tx.traversal().V().has(T.id, P.within(vId, v2Id)).order().by("age").values("age").toList());
+        assertEquals(Arrays.asList(20, 30), tx.traversal().V().has(T.id, P.within(vId, v2Id)).order().by("age").values("age").toList());
+        assertEquals(Arrays.asList(20, 30), tx.traversal().V().hasId(vId, v2Id).order().by("age").values("age").toList());
+        assertEquals(Arrays.asList(20, 30), tx.traversal().V().hasId(vId, v2Id).order().by("age").values("age").toList());
+
+        // FIXME: vertices lookup in Or query
+        // assertEquals(1, tx.traversal().V().or(__.hasId(vId), __.has("pos", 100)).count().next());
```

Comment:
This is another bug I found when fixing this issue. I can reproduce the bug on the master branch. Since this is an unrelated bug, I'd like to create a new issue for it.